docs/options api.md

```diff
@@ -0,0 +1,204 @@
+can use to equate to a specific value. For example:
+```python
+special_range_names: {
+    "normal": 20,
+    "extreme": 99,
+}
+```
+
+will let users use the names "normal" or "extreme" in their options selections,
+but will still return those as integers to you. Useful if you want special
+handling regarding those specified values.
+
+## More Advanced Options
+### FreeText
+This is an option that allows the user to enter any possible string value. Can
+only be compared with strings, and has no validation step, so if this needs
+to be validated, you can either add a validation step to the option class or
+within the world.
+
+### TextChoice
+Like choice allows you to predetermine options and has all of the same comparison
+methods and handling. Also accepts any user defined string as a valid option,
+so will either need to be validated by adding a validation step to the option
+class or within world, if necessary. Value for this class is `Union[str, int]`
+so if you need the value at a specified point,
+`self.multiworld.my_option[self.player].current_key` will always return a string.
+
+### PlandoBosses
+An option specifically built for handling boss rando, if your game can use it.
+Is a subclass of TextChoice so supports everything it does, as long as having
+multiple validation steps to automatically support boss plando from the users.
+If using this class, you must define `bosses`, a set of valid boss names, and
+`locations`, a set of valid boss location names, and `def can_place_boss`, which
+passes a boss and location, allowing you to check if that placement is valid for
+your game. When this function is called, `bosses`, `locations`, and the passed
+strings will all be lowercase. There is also a `duplicate_bosses` attribute
+allowing you to define if a boss can be placed multiple times in your world.
+False by default, and will reject duplicate boss names from the user. For an
+example of using this class refer to `worlds.alttp.options.py`
+
+### OptionDict
+This option returns a dictionary. Setting a default here is recommended as it
+will output the dictionary to the template. If you set a [Schema](https://pypi.org/project/schema/) on the class with
+`schema = Schema()`, then the options system will automatically validate the user supplied data against the schema to
+ensure it's in the correct format.
+
+### ItemDict
+Like OptionDict, except will verify that every key in the dictionary is a valid
+name for an item for your world.
+
+### OptionList
+This option defines a List, where the user can add any number of strings to said
+list, allowing duplicate values. You can define a set of keys in `valid_keys`,
+and a default list if you want certain options to be available without editing
+for this. `valid_keys_casefold` to true will make the verification
+case-insensitive, `verify_item_name` will check that each value is a valid item
+name, and`verify_location_name` will check that each value is a valid location name.
```

Comment:
```suggestion
case-insensitive; `verify_item_name` will check that each value is a valid item
name; and`verify_location_name` will check that each value is a valid location name.
```
Probably better off as a bulleted list, but semicolons break this up more cleanly than colons do.